### ISSUE TITLE
JavaScript/TypeScript: it should be possible for IntelliSense to provide import suggestions in the same way as Quick Open (Ctrl+P)

### ISSUE THREAD

**Issue by jakub-g** [microsoft/vscode#105980]
# Issue description

When I press `Ctrl-P`, I can quickly find any file. It's blazing fast and finds matches in subdirectories etc. 
For example I have a file called `timestamp.js` which is found instantly:
![image](https://user-images.githubusercontent.com/1437027/92040202-b3ff5380-ed76-11ea-99fb-c24d28341622.png)

When I want to import this file in my JS code, it won't be found unless I start typing its exact full path. Otherwise, after a long wait, I will get "no suggestions".
![image](https://user-images.githubusercontent.com/1437027/92040175-a944be80-ed76-11ea-97f6-44736edaee42.png)

The in-editor completion often happens to be randomly very slow for me, and despite disabled extensions, reading through open issues etc. I didn't manage to understand the issue.

# Proposal

I don't know how in-editor path completion works, but it seems to me that it's doing much more work than `Quick Open (Ctrl+P)` (unclear if this is related to parsing the statement, or reading disk, or other things?)
I would gladly prefer a dumb but fast `Quick Open` style completion, rather than environment-aware but slow "smart" completion (maybe behind a flag, or under a different keyboard shortcut than `ctrl-space` from IntelliSense?) 

That is, I wouldn't mind if it suggests me to import a JPEG file, as long as it's as fast as `Ctrl-P`. I know my JS/JSX files names, so it's non-issue for me. But I don't know the deep paths where they are located, and I don't want to wait tens of seconds for the IntelliSense completion to finish.

**TL;DR** I'd prefer _dumb but fast_ path completion over _smart but slow_ if possible (perhaps behind the flag).

**Comment by jasonwilliams** [comment]
If anyone lands here this is being discussed in https://github.com/microsoft/TypeScript/issues/31658
